packages/microservice-util-lib/src/openapi-fetch-middlewares/authentications.ts

```diff
@@ -0,0 +1,106 @@
+import type { Middleware } from 'openapi-fetch';
+import { generateOauthParams } from './oauth10a/oauth10a';
+import type { ApiKey, Basic, OAuth10a, OAuth20 } from './types/authentications';
+
+/**
+ * Creates an openapi-fetch middleware for API key authentication.
+ * This middleware sets the API key in the specified header for each request.
+ *
+ * @param {ApiKey} config - The configuration for API key authentication.
+ * @returns {Middleware} The middleware for API key authentication.
+ *
+ * @example
+ * const middleware = apiKeyAuthMiddleware({
+ *     header: 'x-api-key',
+ *     value: async () => 'your-api-key',
```

Comment:
Any particular reason you have to pass async functions in to the middlewares?

Reply:
Yeah, I pass async function to this so we can be flexible on getting the credentials from SSM (or other secret managers) if needed. This wouldn't be an issue if we support top level await but it's not the time yet :)